compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPULoweringConfigUtils.h

```diff
@@ -28,6 +28,26 @@ void setSubgroupNCount(MLIRContext *context,
                        SmallVectorImpl<NamedAttribute> &attrs,
                        int64_t subgroupNCount);
 
+// Helper to retrieve/set distribution basis.
+//
+// The basis consists of two integer arrays:
+//   - "counts": number of resource to use per dimension in the basis.
+//   - "mapping": a projected permutation to map to basis to the operations
+//     iteration space.
+//
+// Given a resource "x", the "basis" can be used to determine the distribution
+// of an iteration space using:
+//
+// b = delinearize(x, counts)
+// idx = apply(b, mapping)
+LogicalResult getBasis(IREE::GPU::LoweringConfigAttr config,
```

Comment:
Nit: I generally prefer not using return by reference. Its easy to mess up. Maybe have a new struct
```
struct Basis {
  SmallVector<int64_t> basis;
  SmallVector<int64_t> mapping;
};
```
and return `FailureOr<Basis>`. The  `setBasis` method could just take `const Basis &` as an argument

Reply:
done